homeassistant/components/remote/xiaomi_miio.py

```diff
@@ -0,0 +1,255 @@
+            message = yield from hass.async_add_job(
+                device.read, slot)
+            _LOGGER.debug("Message recieved from device: '%s'", message)
+
+            if 'code' in message and message['code']:
+                log_msg = "Received command is: {}".format(message['code'])
+                _LOGGER.info(log_msg)
+                hass.components.persistent_notification.async_create(
+                    log_msg, title='Xiaomi Miio Remote')
+                return
+
+            if ('error' in message and
+                    message['error']['message'] == "learn timeout"):
+                yield from hass.async_add_job(device.learn, slot)
+
+            yield from asyncio.sleep(1, loop=hass.loop)
+
+        _LOGGER.error("Timeout. No infrared command captured")
+        hass.components.persistent_notification.async_create(
+            "Timeout. No infrared command captured",
+            title='Xiaomi Miio Remote')
+
+    hass.services.async_register(DOMAIN, SERVICE_LEARN, async_service_handler,
+                                 schema=LEARN_COMMAND_SCHEMA)
+
+
+class XiaomiMiioRemote(RemoteDevice):
+    """Representation of a Xiaomi Miio Remote device."""
+
+    def __init__(self, friendly_name, device,
+                 slot, timeout, hidden, commands):
+        """Initialize the remote."""
+        self._name = friendly_name
+        self._device = device
+        self._is_hidden = hidden
+        self._slot = slot
+        self._timeout = timeout
+        self._state = False
+        self._commands = commands
+
+    @property
+    def name(self):
+        """Return the name of the remote."""
+        return self._name
+
+    @property
+    def device(self):
+        """Return the remote object."""
+        return self._device
+
+    @property
+    def hidden(self):
+        """Return if we should hide entity."""
+        return self._is_hidden
+
+    @property
+    def slot(self):
+        """Return the slot to save learned command."""
+        return self._slot
+
+    @property
+    def timeout(self):
+        """Return the timeout for learning command."""
+        return self._timeout
+
+    @property
+    def is_on(self):
+        """Return False if device is unreachable, else True."""
+        from miio import DeviceException
+        try:
+            self.device.info()
+            return True
+        except DeviceException:
+            return False
+
+    @property
+    def should_poll(self):
+        """We should not be polled for device up state."""
+        return False
+
+    @property
+    def device_state_attributes(self):
+        """Hide remote by default."""
+        if self._is_hidden:
+            return {'hidden': 'true'}
+        else:
+            return
+
+    # pylint: disable=R0201
+    @asyncio.coroutine
+    def async_turn_on(self, **kwargs):
+        """Turn the device on."""
+        _LOGGER.error("Device does not support turn_on, " +
```

Comment:
Remove `+`, it's not needed.

Reply:
Usually we let not implemented methods raise that error if called. I'd just remove these methods from the class here.